### ISSUE TITLE
array-type doesn't consider "object" to be a simple type

### ISSUE THREAD

**Issue by lfairy** [palantir/tslint#2351]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function foo(a: object[]): object[] {
    return a;
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended",
    "array-type": [true, "array-simple"]
}
```

#### Actual behavior

```
test.ts[1, 17]: Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.
test.ts[1, 28]: Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.
```

#### Expected behavior

No warnings reported, since `object` should be considered a simple type along with e.g. `number` and `string`. See the [TypeScript 2.2 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#object-type).
